src/System.Management.Automation/engine/ExecutionContext.cs

```diff
@@ -1229,7 +1229,9 @@ internal static Assembly LoadAssembly(string name, string filename, out Exceptio
                     error = securityException;
                 }
             }
-            else if (!String.IsNullOrEmpty(name))
+            
```

Comment:
How about we check if the file exists before trying `Assembly.LoadFrom`? So maybe just change `if (!String.IsNullOrEmpty(filename))` to `if (File.Exists(filename))`?